Issue: Build error in JSON CPP for TDK Enabled RDK-B Builds

/mnt/jenkins/workspace/ARRISXB3-Yocto-Build/build-arrisxb3/tmp/work/arm1176jzstb-rdk-linux-uclibceabi/jsoncpp/1.8.4-r0/git/src/lib_json/json_reader.cpp:35:18: error: 'snprintf' is not a member of 'std'
15:33:23 |  #define snprintf std::snprintf

std::snprintf() is only available in C++11, which is not provided by
all compilers. Since the C library snprintf() can easily be used as a
replacement on Linux systems, this patch changes jsoncpp to use the C
library snprintf() instead of C++11 std::snprintf(), fixing the build error
below:

src/lib_json/json_writer.cpp:33:18: error: 'snprintf' is not a member of 'std'

Signed-off-by: Sugasini Velusamy <sugasini@tataelxsi.co.in>


Index: git/src/lib_json/json_reader.cpp
===================================================================
--- git.orig/src/lib_json/json_reader.cpp
+++ git/src/lib_json/json_reader.cpp
@@ -31,9 +31,9 @@
 #elif defined(__ANDROID__) || defined(__QNXNTO__)
 #define snprintf snprintf
 #elif __cplusplus >= 201103L
-#if !defined(__MINGW32__) && !defined(__CYGWIN__)
-#define snprintf std::snprintf
-#endif
+//#if !defined(__MINGW32__) && !defined(__CYGWIN__)
+//#define snprintf std::snprintf
+//#endif
 #endif
 
 #if defined(__QNXNTO__)
Index: git/src/lib_json/json_writer.cpp
===================================================================
--- git.orig/src/lib_json/json_writer.cpp
+++ git/src/lib_json/json_writer.cpp
@@ -57,9 +57,9 @@
 #elif defined(__ANDROID__) || defined(__QNXNTO__)
 #define snprintf snprintf
 #elif __cplusplus >= 201103L
-#if !defined(__MINGW32__) && !defined(__CYGWIN__)
-#define snprintf std::snprintf
-#endif
+//#if !defined(__MINGW32__) && !defined(__CYGWIN__)
+//#define snprintf std::snprintf
+//#endif
 #endif
 
 #if defined(__BORLANDC__)  
